nextroute/schema/input.go

```diff
@@ -7,129 +7,217 @@ import (
-	AlternateStops          *[]string  `json:"alternate_stops,omitempty"`
+	// Capacity of the vehicle.
+	Capacity any `json:"capacity,omitempty"`
+	// StartLevel initial level of the vehicle.
+	StartLevel any `json:"start_level,omitempty"`
+	// StartLocation location where the vehicle starts..
+	StartLocation *Location `json:"start_location,omitempty"`
+	// EndLocation location where the vehicle ends..
+	EndLocation *Location `json:"end_location,omitempty"`
+	// Speed of the vehicle in meters per second.
+	Speed *float64 `json:"speed,omitempty"`
+	// StartTime time when the vehicle starts its route.
+	StartTime *time.Time `json:"start_time,omitempty"`
+	// EndTime latest time at which the vehicle ends its route.
+	EndTime *time.Time `json:"end_time,omitempty"`
+	// MinStops minimum stops that a vehicle should visit.
+	MinStops *int `json:"min_stops,omitempty"`
+	// MinStopsPenalty penalty for not visiting the minimum number of stops.
+	MinStopsPenalty *float64 `json:"min_stops_penalty,omitempty"`
+	// MaxStops maximum number of stops that the vehicle can visit.
+	MaxStops *int `json:"max_stops,omitempty"`
+	// MaxDistance maximum distance in meters that the vehicle can travel.
+	MaxDistance *int `json:"max_distance,omitempty"`
+	// MaxDuration maximum duration in seconds that the vehicle can travel.
+	MaxDuration *int `json:"max_duration,omitempty"`
+	// MaxWait maximum aggregated waiting time that the vehicle can wait across route stops.
+	MaxWait *int `json:"max_wait,omitempty"`
+	// CompatibilityAttributes attributes that the vehicle is compatible with.
+	CompatibilityAttributes *[]string `json:"compatibility_attributes,omitempty"`
+	// ActivationPenalty penalty of using the vehicle.
+	ActivationPenalty *int `json:"activation_penalty,omitempty"`
+	// AlternateStops a set of alternate stops for which only one should be serviced.
+	AlternateStops *[]string `json:"alternate_stops,omitempty"`
 }
 
 // StopDefaults contains default values for stops.
 type StopDefaults struct {
-	UnplannedPenalty        *int       `json:"unplanned_penalty,omitempty"`
-	Quantity                any        `json:"quantity,omitempty"`
-	StartTimeWindow         any        `json:"start_time_window,omitempty"`
-	MaxWait                 *int       `json:"max_wait,omitempty"`
-	Duration                *int       `json:"duration,omitempty"`
-	TargetArrivalTime       *time.Time `json:"target_arrival_time,omitempty"`
-	EarlyArrivalTimePenalty *float64   `json:"early_arrival_time_penalty,omitempty"`
-	LateArrivalTimePenalty  *float64   `json:"late_arrival_time_penalty,omitempty"`
-	CompatibilityAttributes *[]string  `json:"compatibility_attributes,omitempty"`
+	// UnplannedPenalty penalty for not planning a stop.
+	UnplannedPenalty *int `json:"unplanned_penalty,omitempty"`
+	// Quantity of the stop.
+	Quantity any `json:"quantity,omitempty"`
+	// StartTimeWindow time window in which the stop can start service.
```

Comment:
We need to roughly outline which kind of types can be used here. E.g.: "This can be a single window given as a 2-long array of RFC3339 strings (e.g.: `["2023-01-01T12:00:00+01:00", "2023-01-01T13:00:00+01:00"]`). Or, an array of multiple of these windows.